inc/admin/upgrader.php

```diff
@@ -248,5 +248,23 @@ function rocket_new_upgrade( $wp_rocket_version, $actual_version ) {
 		rocket_clean_cache_busting();
 		rocket_generate_advanced_cache_file();
 	}
+
+	if ( version_compare( $actual_version, '2.11.2', '<' ) ) {
+		$options = get_option( WP_ROCKET_SLUG );
+
+		if ( 'http://www.wp-rocket.me' === $options['wl_plugin_URI'] ) {
+			$options['wl_plugin_URI'] = 'https://wp-rocket.me';
+		}
+
+		if ( 'WP Rocket' === $options['wl_author'] ) {
+			$options['wl_author'] = 'WP Media';
+		}
+
+		if ( 'http://www.wp-rocket.me' === $options['wl_author_URI'] ) {
+			$options['wl_author_URI'] = 'https://wp-media.me';
+		}
+
+		update_option( WP_ROCKET_SLUG, $options );
+	}
```

Comment:
You're assuming a lot of things here.
```php		
	if ( version_compare( $actual_version, '2.11.2' ) < 0 ) {
		$options = get_option( WP_ROCKET_SLUG );
		$options = is_array( $options) ? $options : array();
		$update  = false;

		if ( ! isset( $options['wl_plugin_URI'] ) || 'http://www.wp-rocket.me' === $options['wl_plugin_URI'] ) {
			$options['wl_plugin_URI'] = 'https://wp-rocket.me';
			$update = true;
		}

		if ( ! isset( $options['wl_author'] ) || 'WP Rocket' === $options['wl_author'] ) {
			$options['wl_author'] = 'WP Media';
			$update = true;
		}

		if ( ! isset( $options['wl_author_URI'] ) || 'http://www.wp-rocket.me' === $options['wl_author_URI'] ) {
			$options['wl_author_URI'] = 'https://wp-media.me';
			$update = true;
		}

		if ( $update ) {
			update_option( WP_ROCKET_SLUG, $options );
		}
	}
```

Reply:
thank you for that, indeed it's much safer that way